docs/Reference/Server.md

```diff
@@ -1572,6 +1575,35 @@ if (statusCode >= 500) {
   log.error(error)
 }
 ```
+
+### Custom error handler has to account for stream replies.
+
+If `Content-Type` is different between endpoint and error handler
+For example, endpoint returns `application/text` stream and error handler
+responds with `application/json` JSON data. In this case `Content-Type` has
+to be explicitly defined in both endpoint and error handler. Otherwise, the
+error handler would fail serialization with `500` status code as it would not
+apply JSON serialization logic.
+Another option would be to always respond with serialized data in error handler,
+by manually calling serialization method (e.g. `JSON.stringify`).
```

Comment:
```suggestion
##### Custom error handler has to account for stream replies

If `Content-Type` differs between the endpoint and error handler, explicitly
define it in both. For example, if the endpoint returns an `application/text`
stream and the error handler responds with `application/json`, the error handler
must explicitly set `Content-Type`. Otherwise, it will fail serialization with
a `500` status code. Alternatively, always respond with serialized data in the
error handler by manually calling a serialization method (e.g.,
`JSON.stringify`).
```